### PR TITLE
Bump all Go dependencies

### PR DESCRIPTION
**Pull request by lucacome**
Bumps all the dependencies to the latest version available. Hopefully after this dependabot will start behaving...

### CODE REVIEW

**Comment by Jefftree** [comment]
Looks like a type change in gengo: https://github.com/kubernetes/gengo/commit/2a31dd793ca4951ebf250828ad09654bceb47e23.

**Comment by lucacome** [comment]
Reverting gengo so we can update it in a separate PR

**Comment by lucacome** [comment]
@Jefftree can we merge this?

**Comment by Jefftree** [comment]
hey, sorry been super busy with code freeze. Would you mind opening a k/k PR just to make sure tests are passing? I can also do it later this week.

**Comment by lucacome** [comment]
Do you mean after these changes are merged?
If so I already have a PR up that I'm going to update https://github.com/kubernetes/kubernetes/pull/124632

**Comment by Jefftree** [comment]
Before the changes are merged by creating a temp PR pinned to your kube-openapi update branch and verifying the tests have no problems

https://github.com/kubernetes/kubernetes/blob/6a95d2048e3dbb51db0e10a50d021a2e77ce79a3/go.mod#L248

Created the test PR here: https://github.com/Jefftree/kubernetes/pull/new/bump-openapi-test

**Comment by lucacome** [comment]
oh I didn't know you could pin it like that, cool!

**Comment by Jefftree** [comment]
tests look good!
/lgtm
/approve

File: go.mod [hunk 1]
```diff
@@ -4,47 +4,44 @@ go 1.20
...
 	github.com/NYTimes/gziphandler v1.1.1
-	github.com/asaskevich/govalidator v0.0.0-20190424111038-f61b66f89f4a
-	github.com/emicklei/go-restful/v3 v3.11.0
-	github.com/go-openapi/jsonreference v0.20.1
-	github.com/go-openapi/swag v0.22.3
-	github.com/golang/protobuf v1.5.2
+	github.com/asaskevich/govalidator v0.0.0-20230301143203-a9d515a09cc2
+	github.com/emicklei/go-restful/v3 v3.12.1
+	github.com/go-openapi/jsonreference v0.21.0
+	github.com/go-openapi/swag v0.23.0
```
Comment (liggitt) [inline go.mod:10]: govalidator, go-restful, jsonreference need review before bumping, these impact kubernetes API surface and behavior

**Pull request merged**